### PR TITLE
feat(url): parse URLs

### PR DESCRIPTION
**Pull request by Duologic**
Title says it, this adds a function to parse (and join) URLs. I've mostly followed
Python's implementation. This PR also includes several unit tests.

This PR distils the URL parsing from https://github.com/crdsonnet/crdsonnet/pull/11

### CODE REVIEW

File: docs/jsonpath.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ This function does not support  escaping brackets/quotes in path keys.
 ### fn getJSONPath
 
 ```ts
-getJSONPath(source, path)
+getJSONPath(source, path, default='null')
```
Comment (Duologic) [inline docs/jsonpath.md:35]: unrelated change from `make docs`

File: url.libsonnet [hunk 2]
```diff
@@ -4,18 +4,98 @@ local d = import 'doc-util/main.libsonnet';
   '#': d.pkg(
     name='url',
     url='github.com/jsonnet-libs/xtd/url.libsonnet',
-    help='`url` implements URL escaping and query building',
+    help='`url` provides functions to deal with URLs',
   ),
 
-  '#escapeString': d.fn('`escapeString` escapes the given string so it can be safely placed inside an URL, replacing special characters with `%XX` sequences', [d.arg('str', d.T.string), d.arg('excludedChars', d.T.array)]),
+  '#escapeString': d.fn(
```
Comment (Duologic) [inline url.libsonnet:10]: broken docstrings into multiple lines to improve readability

**Review by jvrplmlmn — APPROVED** [review]
👍
